.scripts/pm_apt_install.sh

```diff
@@ -4,38 +4,97 @@ IFS=$'\n\t'
 
 pm_apt_install() {
     local APPNAME=${1:-}
+    local APPDEPENDENCYOF=${2:-}
     local REDIRECT="> /dev/null 2>&1"
     if [[ ${CI:-} == true ]]; then
         REDIRECT="> /dev/null"
     fi
     if [[ ${APPNAME} != "" ]]; then
-        notice "Installing or updating ${APPNAME} via apt"
-        local APP_PACKAGE
-        APP_PACKAGE=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt_name" || true)
-        local SOURCE_FILE="/etc/apt/sources.list.d/${APPNAME,,}.list"
+        local UPDATE_APT
         local SOURCE_REPO
-        SOURCE_REPO=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt_repo" || true)
-        if [[ ${SOURCE_FILE} != "" && ${SOURCE_REPO} != "" ]]; then
-            info "Adding/updating sources for ${APPNAME}"
-            if [[ -f ${SOURCE_FILE} ]]; then
-                rm "${SOURCE_FILE}"
+        SOURCE_REPO=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt.${DETECTED_DISTRO}.${DETECTED_CODENAME}.repo" || true)
+        if [[ ${SOURCE_REPO} == "" ]]; then
+            SOURCE_REPO=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt.${DETECTED_DISTRO}.repo" || true)
+        fi
+        if [[ ${SOURCE_REPO} == "" ]]; then
+            SOURCE_REPO=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt.general.repo" || true)
+        fi
+        if [[ ${SOURCE_REPO} != "" ]]; then
+            if [[ ${SOURCE_REPO} == deb* ]]; then
+                local SOURCE_FILE="/etc/apt/sources.list.d/${APPNAME,,}.list"
+                local SOURCE_EXISTS
+                SOURCE_EXISTS=$(grep -c "${SOURCE_REPO}" "${SOURCE_FILE}" || echo "0")
+                local SOURCE_LINE_COUNT
+                SOURCE_LINE_COUNT=$(cat "${SOURCE_FILE}" | wc -l || echo "0")
```

Comment:
Yeah, that was dumb on my part since wc supports reading files. Will circle back to looking at implementing what you put instead of awk that I updated it to use since wc give the count with the filename after it...